os/src/SubProcess.scala

```diff
@@ -49,19 +49,27 @@ sealed trait ProcessLike extends java.lang.AutoCloseable {
    */
   def waitFor(timeout: Long = -1): Boolean
 
+  // FIXME: docs
   /**
    * Wait up to `millis` for the [[ProcessLike]] to terminate and all stdout and stderr
    * from the subprocess to be handled. By default waits indefinitely; if a time
    * limit is given, explicitly destroys the [[ProcessLike]] if it has not completed by
    * the time the timeout has occurred
    */
-  def join(timeout: Long = -1): Boolean
+  def join(timeout: Long = -1, timeoutGracePeriod: Long = 1000): Boolean
+
```

Comment:
The current logic for `join` (undocumented) returns `true` if the process had already terminated by the time the timeout completes. If the process is required to terminate in anyway, this will be `false`.

Would it be better instead to have `exitedCleanly` also mean "exited within the grace period and accepted `SIGTERM` without needing `SIGKILL`"? I would say that also feels like a "clean exit", in that no forcible termination was required.

Reply:
I think the current behavior is probably fine; especially for backwards compat we should leave it in place. If people want more control over exactly what happens in the scenario where things exit during the grace period, they can always wrap os.proc.spawn` themselves

Reply:
Ok! I'll make sure that this behaviour is clearly documented then